lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -105,4 +114,31 @@ public function getLastResponse($force = false)
 
         return ($force) ? $lastResponse : $this->lastCachedResponse;
     }
+
+    /**
+     * Format the path and add query parameters if they exist.
+     *
+     * @param string $path
+     * @param array  $options
+     * @return void
+     */
+    private function formatPath($path, array $options)
+    {
+        $this->path = $path;
+
+        if (array_key_exists('query', $options) && !empty($options['query'])) {
+            $this->path .= '?';
+
+            $i = 0;
+            foreach ($options['query'] as $key => $value) {
```

Comment:
@stof I finally got my error. I used `$client->getHttpClient()->get(...);` the wrong way!
The signature is of course `public function get($path, array $parameters = array(), array $headers = array())` so I and everybody have to pass query parameters to the array `$parameters`...

So of course it simplifies the problem now.

For both custom and regular queries Guzzle will take care of encoding query parameters if they exist as it already does in the original code. So in any case we can fetch them by using the `$request->getQuery()` method.

**We just have to update the `$path` variable and concatenate the encoded query parameters if they exist. No need to class property. No need to extra method...**
You should be happy ;)

Let me update the code and give me your feedback again.

Hope we will be able to merge then.